src/lib/support/ReferenceCountedHandle.h

```diff
@@ -25,30 +25,41 @@ template <typename Target>
 class ReferenceCountedHandle
 {
 public:
-    explicit ReferenceCountedHandle(Target & target) : mTarget(target) { mTarget.Retain(); }
+    explicit ReferenceCountedHandle(Target & target) : mTarget(&target) { mTarget->Retain(); }
+    ~ReferenceCountedHandle()
+    {
+        if (mTarget != nullptr)
+        {
+            // release only if target has not been moved out
+            mTarget->Release();
+        }
+    }
 
-    // Ideally we would suppress this from within Optional.h, where this false positive is coming from. That said suppressing
-    // here is okay since no other cases could create instance of ReferenceCountedHandle without going through explicit
-    // contstructor.
-    //
-    // NOLINTNEXTLINE(clang-analyzer-core.CallAndMessage): Only in a false positive is mTarget uninitialized.
-    ~ReferenceCountedHandle() { mTarget.Release(); }
+    ReferenceCountedHandle(const ReferenceCountedHandle & that) : mTarget(that.mTarget) { mTarget->Retain(); }
+    ReferenceCountedHandle(ReferenceCountedHandle && that) : mTarget(std::move(that.mTarget)) { that.mTarget = nullptr; }
```

Comment:
This class should arguably not have a move constructor.  Certainly its previous "move constructor" just did a copy.  Doing what this is doing breaks the "this is immutable" guarantee about the mTarget member.